docs/specs/editorconfig-in-the-compiler.md

```diff
@@ -0,0 +1,110 @@
+
+*This section is informational only and does not propose any new behavior.*
+
+As a reminder, the `/pathmap` flag to the compiler allows the compiler to "rewrite" paths prior to them being
+seen in the following cases today:
+
+- The embedding of the path into a PDB file.
+- The embedding of the path into a `[CallerFilePath]` argument.
+
+The syntax for this flag is of the form:
+
+```
+    /pathmap:from1=to1,from2=to2,...
+```
+
+# Computation of resultant .editorconfig properties for a given file
+
+The computation of the resultant .editorconfig values for a given file *MUST* give the same output as the algorithm
+below. Actual implementations may break these into different stages and may have some caching/optimizations as appropriate,
+but such caching *MUST NOT* be observable.
+
+1. The compiler goes through and combines any non-absolute-pathed /editorconfig: switches, and combines them the current
+   working directory of the compiler. Parent folder references are eliminated; for example `C:\Dog\Cat\..\Dog\.editorconfig` is converted to `C:\Dog\Dog\.editorconfig`.
+   All casing is preserved during this transformation. The order of .editorconfig files as given is preserved to the compiler.
+   Call this resulting list `editor_config_files`
+2. For each `source_file_path` given to the compiler (applying to source files and additional files), we:
+   1. If `source_file_path` is a relative path, convert it to an absolute path relative to the current working directory of the compiler.
+      Parent folder references are eliminated the same as we did in step #1 for .editorconfig files. Call this `source_full_file_path`.
+      If the path is already absolute, we still do parent folder reference elimination and call this `source_full_file_path`. Casing is preserved.
+   2. Take `source_full_file_path` and map it through any pathmaps, producing `source_full_mapped_path`.
+   3. Define a dictionary `editorconfig_properties`, a map of string to string, as empty for this file.
+   4. Loop through each `editor_config_file` in `editor_config_files`:
+      1. Strip the filename portion of `editor_config_file` and determine if the resulting directory is a container of
+         `source_full_mapped_path`. This determination is simply if each path component of `editor_config_file`'s directory
+         name is present, in order and case sensitively, at the start of `source_full_mapped_path`. If this check fails,
+         continue to the next `editor_config_file`.
+      2. Compute the relative path `relative_path` of `source_full_mapped_path` to `editor_config_file`. This is in effect the path components
+         that were not looked at in the previous step.
+      3. Determine all sections of the .editorconfig that apply to `relative_path` per the globbing algorithm as defined in the .editorconfig spec.
+      4. For each section, take the defined properties in include them into `editorconfig_properties`, preserving any key/value pairs already in 
```

Comment:
:memo: It's important that this step work from the last section defined in the file to the first section.